cpp/src/arrow/util/CMakeLists.txt

```diff
@@ -54,7 +54,11 @@ endif()
 
 ADD_ARROW_TEST(bit-util-test)
 ADD_ARROW_TEST(checked-cast-test)
-ADD_ARROW_TEST(compression-test)
+if(ZLIB_SHARED_LIB)
+  ADD_ARROW_TEST(compression-test EXTRA_LINK_LIBS ${ZLIB_SHARED_LIB})
+else()
+  ADD_ARROW_TEST(compression-test EXTRA_DEPENDENCIES zlib)
+endif()
```

Comment:
Shouldn't this be exposed as a transitive dependency of arrow_static/arrow_shared? I think there's some way to get CMake to automatically include transitive link dependencies